Rakefile

```diff
@@ -1,25 +1,48 @@
-def xcpretty(cmd)
-  if `which xcpretty` && $?.success?
+def xcpretty(cmd, name)
+  if ENV['CI']
+    sh "set -o pipefail && #{cmd} | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif `which xcpretty` && $?.success?
     sh "set -o pipefail && #{cmd} | xcpretty -c"
   else
     sh cmd
   end
 end
 
-task :build_for_testing do
-  sh "swift build"
-  xcpretty "xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests build-for-testing"
+def plain(cmd, name)
+  if ENV['CI']
+    sh "set -o pipefail && #{cmd} | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\""
+  else
+    sh cmd
+  end
 end
 
-desc 'Run Unit Tests'
-task :test => :build_for_testing do
-  sh "swift test"
-  xcpretty "xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests test-without-building"
+namespace :spm do
+  desc 'Build using SPM'
+  task :build do
+    plain("swift build", "spm_build")
+  end
+
+  desc 'Run SPM Unit Tests'
+  task :test => :build do
+    plain("swift test", "spm_build")
```

Comment:
I tried using a ruby trick to get the calling method's name, but keep getting `block (2 levels) in <top (required)>`, so gave up on that.